compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeTransformer.java

```diff
@@ -4175,19 +4175,28 @@ private BLangMatchPattern transformMatchPattern(Node matchPattern) {
                     (BLangListMatchPattern) TreeBuilder.createListMatchPattern();
             bLangListMatchPattern.pos = matchPatternPos;
 
-            for (Node memberMatchPattern : listMatchPatternNode.matchPatterns()) {
-                BLangMatchPattern bLangMemberMatchPattern = transformMatchPattern(memberMatchPattern);
+            SeparatedNodeList<Node> matchPatterns = listMatchPatternNode.matchPatterns();
+            int matchPatternListSize = matchPatterns.size();
+
+            if (matchPatternListSize < 1) {
```

Comment:
Isn't this just 
```suggestion
            if (matchPatternListSize == 0) {
```

IMO, it's better to have the exact check, or it could get us wondering if it can be anything other than 0.

Reply:
Okay